shared/reducers/tracker2.tsx

```diff
@@ -109,20 +110,22 @@ export default Container.makeReducer<Actions, Types.State>(initialState, {
     })
   },
   [EngineGen.keybase1NotifyTrackingNotifyUserBlocked]: (draftState, action) => {
-    const {blocker, chatBlocked, followBlocked} = action.payload.params.b
+    const {blocker, blocks} = action.payload.params.b
     const d = getDetails(draftState, blocker)
-    const toProcessChat = (chatBlocked ?? []).map(
-      username => [username, getDetails(draftState, username)] as const
+    const toProcess = Object.entries(blocks ?? {}).map(
+      ([username, userBlocks]) => [username, getDetails(draftState, username), userBlocks || []] as const
     )
-    const toProcessFollow = (followBlocked ?? []).map(
-      username => [username, getDetails(draftState, username)] as const
-    )
-    toProcessChat.forEach(([_, det]) => {
-      det.blocked = true
-    })
-    toProcessFollow.forEach(([username, det]) => {
-      det.hidFromFollowers = true
-      d.followers && d.followers.delete(username)
+    toProcess.forEach(([username, det, userBlocks]) => {
+      userBlocks.forEach(blockState => {
+        if (blockState.blockType === RpcTypes.UserBlockType.chat) {
+          det.blocked = blockState.blocked
+        } else if (blockState.blockType === RpcTypes.UserBlockType.follow) {
+          det.hidFromFollowers = blockState.blocked
+          blockState.blocked
+            ? d.followers && d.followers.delete(username)
+            : d.followers && d.followers.add(username)
```

Comment:
the person might not actually follow you, we shouldn't add them.

Reply:
Hmm, that's a good point. We should add them though if they _are_ followed, right?

Reply:
It's likely you will load your followers again when landing on your profile. Otherwise I think we'd need another notification to add them back to your followers?

Reply:
I wonder if I can pigeonhole it into this gregor...

Reply:
Tempted to merge as-is (with your suggested change), then deal with it separately.

Reply:
Yes seems separate from this